### PR TITLE
[Yul] remove PC instruction

### PR DESCRIPTION
**Pull request by christianparpart**

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Breaking changes:
  * Type Checker: Disallow shifts by signed types.
 
 Language Features:
+ * Yul: Removed EVM instruction `pc()`.
```
Comment (chriseth) [inline Changelog.md]: disallow?

File: libyul/AsmAnalysis.cpp [hunk 2]
```diff
@@ -531,7 +531,7 @@ bool AsmAnalyzer::warnOnInstructions(evmasm::Instruction _instr, SourceLocation
 			+ "\" instruction is " +
 			vmKindMessage +
 			" VMs " +
-			" (you are currently compiling for \"" +
```
Comment (chriseth) [inline libyul/AsmAnalysis.cpp]: Can you please fix this in develop?

File: libyul/AsmAnalysis.cpp [hunk 3]
```diff
@@ -557,13 +557,7 @@ bool AsmAnalyzer::warnOnInstructions(evmasm::Instruction _instr, SourceLocation
...
 		errorForVM("only available for Istanbul-compatible");
 	else if (_instr == evmasm::Instruction::PC)
-		m_errorReporter.warning(
-			2450_error,
-			_location,
-			"The \"" +
-			boost::to_lower_copy(instructionInfo(_instr).name) +
-			"\" instruction is deprecated and will be removed in the next breaking release."
-		);
+		errorForVM("not available in Yul for any");
```
Comment (chriseth) [inline libyul/AsmAnalysis.cpp]: Can you make the error similar to the one for jump?

**Comment by mijovic** [comment]
Some tests need to be fixed.

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
This didn't remove the `pc()` from `libyul/backends/wasm/EVMToEwasmTranslator.cpp`.

**Pull request merged**